rest/data/project_settings.go

```diff
@@ -256,6 +260,15 @@ func SaveProjectSettingsForSection(ctx context.Context, projectId string, change
 			}
 		}
 
+		config, err := evergreen.GetConfig()
+		if err != nil {
+			return nil, errors.Wrap(err, "getting evergreen config")
+		}
+		_, err = model.ValidateEnabledProjectsLimit(projectId, config, *mergedSection)
```

Comment:
should validateEnabledProjectsLimit take in the original ref as well so we don't re-query mergedbeforeRef? (for the create case can pass in an empty project ref, or could refactor that function so there are two different functions that reuse a lot of the code)